Source/JavaScriptCore/wasm/js/JSToWasm.cpp

```diff
@@ -207,13 +211,301 @@ void marshallJSResult(CCallHelpers& jit, const TypeDefinition& typeDefinition, c
         jit.setupArguments<decltype(operationAllocateResultsArray)>(GPRInfo::wasmContextInstancePointer, CCallHelpers::TrustedImmPtr(&typeDefinition), indexingType, savedResultsGPR);
         JIT_COMMENT(jit, "operationAllocateResultsArray");
         jit.callOperation<OperationPtrTag>(operationAllocateResultsArray);
+        exceptionChecks.append(jit.branchTestPtr(CCallHelpers::NonZero, GPRInfo::returnValueGPR2));
```

Comment:
Since we don't have a VM readily available here it makes sense that we rely on the exception register but in the future I think it would be better to do 
```
static_assert(operationExceptionRegister<operationAllocateResultsArray>() != InvalidGPRReg, "We don't have a VM readily available so we rely on exception being returned");
exceptionChecks.append(jit.branchTestPtr(CCallHelpers::NonZero, operationExceptionRegister<operationAllocateResultsArray>()));
``` 
That way if the signature changes in the future this doesn't subtly fail, instead, it doesn't even compile.